assets/js/modules/analytics/hooks/useExistingTagEffect.js

```diff
@@ -28,97 +28,40 @@ import { useEffect } from '@wordpress/element';
+	const { setCanUseSnippet, setUseSnippet } = useDispatch(
 		MODULES_ANALYTICS
 	);
-	const gtmModuleActive = useSelect( ( select ) =>
-		select( CORE_MODULES ).isModuleActive( 'tagmanager' )
-	);
-
-	const {
-		existingTag,
-		existingTagAccountID,
-		existingTagPermission,
-		gtmAnalyticsPropertyID,
-		gtmAnalyticsAccountID,
-		gtmAnalyticsPermission,
-	} = useSelect( ( select ) => {
-		const data = {
-			existingTag: select( MODULES_ANALYTICS ).getExistingTag(),
-			existingTagPermission: false,
-			existingTagAccountID: '',
-			gtmAnalyticsPropertyID: '',
-			gtmAnalyticsAccountID: '',
-			gtmAnalyticsPermission: false,
-		};
 
-		if ( data.existingTag ) {
-			// Just check existing tag permissions, if it is available and ignore tag manager settings.
-			const { permission = false, accountID = '' } =
-				select( MODULES_ANALYTICS ).getTagPermission(
-					data.existingTag
-				) || {};
-			if ( permission ) {
-				data.existingTagPermission = permission;
-				data.existingTagAccountID = accountID;
-			}
-		} else {
-			// There is no existing tag, so we need to try to get a property ID from GTM.
-			data.gtmAnalyticsPropertyID = select(
-				MODULES_TAGMANAGER
-			).getSingleAnalyticsPropertyID();
-			if ( data.gtmAnalyticsPropertyID ) {
-				const { permission = false, accountID = '' } =
-					select( MODULES_ANALYTICS ).getTagPermission(
-						data.gtmAnalyticsPropertyID
-					) || {};
-				if ( permission ) {
-					data.gtmAnalyticsAccountID = accountID;
-					data.gtmAnalyticsPermission = permission;
-				}
-			}
-		}
-
-		return data;
-	} );
+	const hasExistingTag = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).hasExistingTag()
+	);
+	const existingTag = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getExistingTag()
+	);
+	const propertyID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getPropertyID()
+	);
 
 	useEffect( () => {
-		if ( existingTag ) {
-			// Disable the plugin snippet
-			setUseSnippet( false );
-			if ( existingTagPermission && existingTagAccountID ) {
-				// There is an existing Analytics tag, select it.
-				setAccountID( existingTagAccountID );
-				selectProperty( existingTag );
-			}
-		} else if (
-			gtmModuleActive &&
-			gtmAnalyticsPropertyID &&
-			gtmAnalyticsPermission &&
-			gtmAnalyticsAccountID
+		if (
+			hasExistingTag &&
+			propertyID &&
+			existingTag &&
+			propertyID === existingTag
 		) {
-			// GTM container has GA tag and user has access to it, force select it.
-			setAccountID( gtmAnalyticsAccountID );
-			selectProperty( gtmAnalyticsPropertyID );
-			// We no longer set `useSnippet` to false here as this is handled using a filter
-			// on the server side now so that Analytics will take over the snippet again
-			// if Tag Manager or its snippet is disabled.
+			// Disable the Analytics snippet if there is an existing tag that
+			// matches the currently selected property.
+			setUseSnippet( false );
+			setCanUseSnippet( false );
```

Comment:
Why the `setCanUseSnippet` here? That is a different option to enforce the value of `useSnippet`. We don't set that today, and it shouldn't happen here either. For example, if the `UseSnippetSwitch` toggle is present in the UI, it shouldn't be disabled in such a case, we should only change its value (via `setUseSnippet`).

Reply:
Ah, okay, I thought we would disable it only _if_ the property matched. Instead we'll just disable it, sounds good 👍🏻